DD/wwwroot/Scripts/chatWidget.js

```diff
@@ -1,96 +1,154 @@
+            data.messages.forEach(message => addMessage(message.content, message.role));
+            return fetchRecentThreads();
+        })
+        .then(() => {
+            currentThreadId= recentThreads[0].id;
+            updateThreadDisplay();
+        })
+        .catch(handleError)
+        .finally(() => showLoading(false));
     }
 
-    async function sendMessage() {
+    function sendMessage() {
         const message = chatInput.value.trim();
         if (!message) return;
 
         addMessage(message, 'user');
         chatInput.value = '';
-        setLoading(true);
-        showTypingIndicator();
-
-        try {
-            const response = await fetch(`${apiBaseUrl}/api/chat/Chat`, {
-                method: 'POST',
-                headers: { 'Content-Type': 'application/json' },
-                body: JSON.stringify({ threadId, userMessage: message }),
-            });
-            if (!response.ok) {
-                throw new Error('Failed to send message');
-            }
-            const data = await response.json();
-            hideTypingIndicator();
-            addMessage(data.response, 'bot');
-        } catch (error) {
-            console.error('Error sending message:', error);
-            hideTypingIndicator();
-            addErrorMessage('Failed to send the message. Please try again.');
-        } finally {
-            setLoading(false);
+        setWaitingForResponse(true);
+
+        const currentThread = recentThreads.find(thread => thread.id === currentThreadId);
+        if (!currentThread) {
+            handleError(new Error('Current thread not found'));
+            setWaitingForResponse(false);
+            return;
         }
+
+        fetch(`${apiBaseUrl}/api/chat/Chat`, {
+            method: 'POST',
+            headers: { 
+                'Content-Type': 'application/json',
+                'Authorization': token
+            },
+            body: JSON.stringify({ threadId: currentThread.threadId, userMessage: message }),
+        })
+        .then(handleResponse)
+        .then(data => {
+            addMessage(data.response, 'assistant');
+            updateThreadDisplay();
+        })
+        .catch(handleError)
+        .finally(() => setWaitingForResponse(false));
+    }
+
+    function loadThread(threadId) {
+        if (threadId === currentThreadId) return;
+
+        showLoading(true);
+        fetch(`${apiBaseUrl}/api/chat/threads/${threadId}/messages`, {
+            headers: { 'Authorization': token }
+        })
+        .then(handleResponse)
+        .then(messages => {
+            currentThreadId = threadId;
+            chatMessages.innerHTML = '';
+            messages.forEach(message => addMessage(message.content, message.role));
+            updateThreadDisplay();
+        })
+        .catch(handleError)
+        .finally(() => showLoading(false));
+    }
+
+    function fetchRecentThreads() {
+        return fetch(`${apiBaseUrl}/api/chat/threads/recent/10`, {
```

Comment:
make the full path dynamic and the number of threads parametrized

Reply:
Added in the latest commit